controllers/role.js

```diff
@@ -0,0 +1,42 @@
+import { Role } from '../models'
+
+const index = async (req, res) => {
+    const tests = await Role.find({});
+    res.status(200).json(tests);
+};
+
+const add = async (req, res) => {
+    const newRole = new Role(req.value.body);
+    const role = await newRole.save();
+    res.status(201).json(role);
+};
+
+const get = async (req, res) => {
+    const { id } = req.value.params;
+    const role = await Role.findById(id);
+    res.status(200).json(role);
+};
+
+const replace = async (req, res) => {
+    const { id } = req.value.params;
+    const newRole = req.value.body;
+    await Role.findByIdAndUpdate(id, newRole);
+    res.status(200).json({ success: true });
+};
+
+const update = async (req, res) => {
+    const { id } = req.value.params;
+    const newRole = req.value.body;
+    await Role.findByIdAndUpdate(id, newRole);
+    res.status(200).json({ success: true });
+};
+
+const switchActivity = async (req, res) => {
+    const { id } = req.value.params;
+    const isActive = req.params[0] === "";
```

Comment:
Maybe it will be better to check actual state of activation

Reply:
That may be useful in some cases, but since it is a PATCH request, feedback on the previous state is not neccessary. And in this situation checking param prefix helps our code to be more extensible, as we need only a single route declaration

Reply:
Could you use util method to check if param is empty? And please use destructuring to avoid using zero index